src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -74,17 +74,18 @@ public async Task<IActionResult> StartDeploymentSession(StartDeploymentSessionIn
             Guid.NewGuid().ToString()
             );
 
-            var serviceProvider = CreateSessionServiceProvider(output.SessionId, input.AWSRegion);
-            var awsResourceQueryer = serviceProvider.GetRequiredService<IAWSResourceQueryer>();
-
             var state = new SessionState(
                 output.SessionId,
                 input.ProjectPath,
                 input.AWSRegion,
-                (await awsResourceQueryer.GetCallerIdentity(input.AWSRegion)).Account,
                 await _projectParserUtility.Parse(input.ProjectPath)
                 );
 
+            var serviceProvider = CreateSessionServiceProvider(state);
+            var awsResourceQueryer = serviceProvider.GetRequiredService<IAWSResourceQueryer>();
+
+            state.AWSAccountId = (await awsResourceQueryer.GetCallerIdentity(input.AWSRegion)).Account;
```

Comment:
Just curious, why move out of the constructor?

Reply:
Before the `serviceProvider` could be created before the session state but now the session state has to be created first in order to create the `serviceProvider` which would allow injecting the `SessionAWSResourceQueryer` into the service collection. 

Since we don't have an `IAWSRequesourceQueryer` used to get the account id at the time of creating the session state I had to move the `AWSAccountId` to being set after the creation of both the session state and the `serviceProvider`.